Mage.Sets/src/mage/cards/s/Soulshriek.java

```diff
@@ -0,0 +1,77 @@
+package mage.cards.s;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.common.delayed.AtTheBeginOfNextEndStepDelayedTriggeredAbility;
+import mage.abilities.dynamicvalue.common.CardsInControllerGraveyardCount;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.SacrificeTargetEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.filter.FilterCard;
+import mage.filter.predicate.mageobject.CardTypePredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetControlledCreaturePermanent;
+import mage.target.targetpointer.FixedTarget;
+
+/*
+ *
+ * @author Ketsuban
+ */
+public class Soulshriek extends CardImpl {
+
+    protected static final FilterCard filterCard = new FilterCard();
+
+    static {
+        filterCard.add(new CardTypePredicate(CardType.CREATURE));
+    }
+
+    public Soulshriek(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId,setInfo,new CardType[]{CardType.INSTANT},"{B}");
+
+        // Target creature you control gets +X/+0 until end of turn, where X is the number of creature cards in your graveyard. Sacrifice that creature at the beginning of the next end step.
+        this.getSpellAbility().addEffect(new SoulshriekEffect());
+        this.getSpellAbility().addTarget(new TargetControlledCreaturePermanent());
+    }
+
+    @Override
+    public Card copy() {
+        return null;
+    }
+}
+
+class SoulshriekEffect extends OneShotEffect {
+
+    public SoulshriekEffect() {
+        super(Outcome.DestroyPermanent);
+        this.staticText = "Target creature you control gets +X/+0 until end of turn, where X is the number of creature cards in your graveyard. Sacrifice that creature at the beginning of the next end step";
+    }
+
+    public SoulshriekEffect(final SoulshriekEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public SoulshriekEffect copy() {
+        return new SoulshriekEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = game.getPermanent(source.getFirstTarget());
+        if (permanent != null) {
+            permanent.addPower(new CardsInControllerGraveyardCount(Soulshriek.filterCard).calculate(game, source, null));

```

Comment:
Did you test this? It needs to be done by adding a continuous effect.

Reply:
When I was looking for similar cards to work off what I ended up with was [Runechanter's Pike](https://github.com/magefree/mage/issues/5716). I wasn't willing to implement bad rules text even if the effect is functional, and I couldn't work out what the problem was with the Pike so I looked for another way of doing it. I assumed this was fine because it's not an effect that needs to be continuously updated - the value of X is checked at most twice, on cast and on resolution.

Reply:
The problem is that it's being done in a OneShotEffect, so the change of power won't last at all. You have to create a `BoostTargetEffect`, set its target to the permanent, then use `game.addEffect` to add it. Try testing it as it is.